github/Branch.py

```diff
@@ -297,7 +299,7 @@ def edit_required_status_checks(
         self,
         strict: Opt[bool] = NotSet,
         contexts: Opt[list[str]] = NotSet,
-    ) -> None:
+    ) -> BranchProtection:
```

Comment:
It looks like the Github API returns `RequiredStatusChecks` here: https://docs.github.com/en/rest/branches/branch-protection?apiVersion=2022-11-28#delete-branch-protection